### PR TITLE
[WIP] Adds Senior Roles into three departments

### PR DESCRIPTION
**Pull request by SamHPurp**
Work in progress, still bits that need to be changed/tweaked - but as something like this'll take yonks to get implemented, it'd be a good idea to get the criticisms from The Big Four, so I can address them.

---------------

**The idea:**

Add three roles one each to Engineering, Security and Medical called "**Senior Engineer**","**Senior Consultant**" and "**Senior Officer**" respectively.

**The details:**

- Functionally identical to a regular Doctor, Engineer or Officer.
- Heavily encouraged to take on interns from the Civilian pool
- A type of IC mentor, available to help members of the department
- Available to step in if the heads of staff are otherwise incapacitated
- In no way command of anything, and neither are they the defacto Acting, but are encouraged to use them
- Access:
- - Currently the idea is to karma lock it. See FAQs

**FAQ:**

**Why Karma locked?**

This ensures that this role only goes to those with experience in the department and the understanding and willingness to train, coach and take on Junior members into the department, 'interns'. They are for all intents and purposes just a regular member of the department, someone paying the karma for what is effectively and 'alt title' - they are dedicated members of that department.

**Why not use the Department tracking system?**

Well firstly, it isn't merged yet, secondly, I like that idea, and it is one certainly to explore, though I still like the idea of karma for the reasons stated above.

**Will this detract from the number of slots in the department?**

Nope, quite the opposite. These roles would be 'added on top' - and on top of that, they are encouraged to get the HoP to assign Civilians/Assistants for them to manage and train. Engineering Intern/Trainee titles allows the crew to know as well that they aren't necessarily to expect 'supermen', or super competence.

**Is this not going to encourage snobbery?**

These roles are ones that is designed to help and aid others - even hiring newbies from the Civ pool - if the situation allows it. As such they will be held to a higher standard.

**Shouldn't mentors get these roles for free?**

No. Lets not blur the lines between OOC mentoring and IC mentoring.

--------------

**Key discussion points**

- Access restrictions: Karma? Playtime? How much of each?
- Should they get anything different? Slightly different beret?

---------------

:cl: Purpose2
add: Adds 'Senior Engineer', 'Senior Consultant' and 'Senior Officer' to their respective departments as Karma roles.
/ :cl:

---------------

Had very positive feedback thus far on the forums where deeper debate has already been raging: https://nanotrasen.se/forum/topic/9900-senior-roles/

### CODE REVIEW

**Comment by Fruerlund** [comment]
Can we get some nice hats or berets sprites for them. Eg. Customs Officer. 

Also 👌

**Comment by lolexploit** [comment]
Does this mean these departments will get a +1 capacity at roundstart?

**Comment by SamHPurp** [comment]
@lolexploit Yes

**Comment by Aurorablade** [comment]
This seems me. Anyone can be an ic mentor. And may lead to entitlement issues. We already have chain of command so for heads. Due to the expectations of this role it gonna need to be policed by admins like a head would be.

**Comment by Aurorablade** [comment]
Meh not me

**Comment by SamHPurp** [comment]
@Aurorablade They are not in the chain of command at all. Think of them as Engineers/Doctors that Nanotrasen has had on staff long enough, they've threatened to quit and someone in HR offered them a job title change in order to get them to stay. Nothing else is different really. They are however encouraged to pass their knowledge on, hence my references to taking on interns.

Couple of admins I pitched this to, said that they didn't see it increasing any of their work load, and one of them mentioned that more people trying to act to a higher standard could "only be a good thing".

**Comment by Crazylemon64** [comment]
Shortly before this is merged, could you make map edits to metastation, too?

**Comment by SamHPurp** [comment]
@Crazylemon64 I'll make these changes to Metastation now.

I'd appreciate your weigh in, if you have time, on the access restriction method... Karma? Playtime? How much of either?

**Comment by Crazylemon64** [comment]
I think both Karma and Playtime will be effective means of gating this - though I'm not sure on the specific values to use. Perhaps 5?

**Comment by Aurorablade** [comment]
You are pitching it as both a mentor and someone who takes over for a head. Id they are taking over for a head THEY HAVE TO  fall under chain of command. Plus because being able to hire civs both kinda takes away from the hop and I would say risk of metacliques but then with karma and play restrict it may limit that.

**Comment by SamHPurp** [comment]
@Aurorablade please re-read the post.

> and neither are they the defacto Acting, but are encouraged to use them

They aren't members of command in any way. They aren't even who gets put as Acting if the Head dies. Captain or HoP can do whomever they like.

They cannot hire themselves, the CE still needs to approve transfers, and the HoP has to action it, as normal. So I don't think thats really a threat either.

**Comment by Aurorablade** [comment]
You made it sound otherwise and I did read the post. A few times. Still this may put people.in an odd mindset in the role but eh.

**Comment by SamHPurp** [comment]
@Crazylemon64 Updated with Metastation.

Concerned five might be too low of a barrier, but as long as there is departmental age tracking via @Kyep 's thing, I think thats totally valid.

**Comment by sothangel** [comment]
Give them something unique to identify them aside from the ID (Armband or something that attaches to jumpsuit or a fancy hat as been said before) and it'd be good.
I'm not really sure what to suggest the karma value be, I would say 10 but this has absolutely no reasoning behind it; just that it sounds like a nice number to have.

**Comment by Regen1** [comment]
Agree that a sort of armband would be a good identifier.

**Comment by SamHPurp** [comment]
@Regen1 its one per department, currently.

**Comment by uraniummeltdown** [comment]
We have mentors to help people out, don't really feel this is necessary. Most people will if not antag help you out if you need help with something, then there's the wiki and mentors. A good head of staff will always help a newbie out anyway. This will make Command prefer Seniors taking Acting Head positions which is paying karma for an IC advantage, very useful for antagging as Acting CE/CMO. Security is not really a mechanically complex department that a Senior role would fit.

**Comment by SamHPurp** [comment]
> 👎 We have mentors to help people out, don't really feel this is necessary. Most people will if not antag help you out if you need help with something, then there's the wiki and mentors.

Somethings you just have to show someone in game. Also not everyone uses that feature, and theres lots of stuff you wouldn't/couldn't tell people about through mhelps. This isn't a system to replace mentors in any way.

>A good head of staff will always help a newbie out anyway. 

Heads of staff have a lot of stuff going on. Yes we're always going to help if we can. Plus people keep talking about there being loads of administrative stuff that Heads aren't doing. This helps free them up a bit.

>This will make Command prefer Seniors taking Acting Head positions which is paying karma for an IC advantage, very useful for antagging as Acting CE/CMO.

This is just straight a non-issue - It specifically states they aren't the defacto acting. Also this doesn't change human nature of what currently happening. If I'm Engineer, not CE, I get picked for Acting CE over others, because the Captain knows he can trust me.

>Security is not really a mechanically complex department that a Senior role would fit.

Thread after thread after thread disagrees with you.

**Comment by ExitGame** [comment]
Am mentor and can approve. Sometimes when people ask in Mhelp about something like "How do I do Virology?" or "I'm new to Medical Doctor." It's really hard to answer those questions without showing them around as a IC character.

**Comment by uraniummeltdown** [comment]
@SamHPurp Not everyone using mhelp is not an excuse, if someone needs help and they don't try mhelp it's their own fault. Sec is not a mechanically complex job, you didn't provide an argument as to why you think it is. Follow space law, click evildoer, use comms. Captains always picking you for acting CE over others is metashit we shouldn't encourage, it's like "oh I don't trust X he's always antag".

@ExitGame 
We have the wiki for that though, the wiki has guides for every job. Do we really need a job solely to hold the hands of players too lazy/young to read a wiki page?

We have a wiki, discord help channel, mentorhelp that are perfectly fine. This is completely unnecessary

**Comment by alexkar598** [comment]
I don't like the idea of karma locking it since karma is really hard to get these days.

**Comment by Fox-McCloud** [comment]
This isn't going to be remotely like what it's being presented as. 

It's just going to become "ha, I paid karma for a special treatment regular job slot."

There's going to be plenty of people (myself included) who are just going to use this as a way to get past the job limit of a job, then proceed to play it like a regular job and not bother helping out anyone, ever, then be the "shoe in" for the head role the moment the other one goes rogue/kicks the bucket.

This is even less fleshed out than the coroner job, as it's literally the same exact job as the default only it has non-tangible reasons to do something different with the job.

This is also going to simultaneously encourage people who don't want to teach anyone to do so even less; "stop bothering me; it's not my job to help you, bother the senior.".

This job won't change anything in terms of "helping" people; those who wanted to help people out while doing their jobs will continue to do so, and those who have no interest will end up picking up this job anyway and using it as "standard department job+".

**Comment by Kyep** [comment]
@Fox-McCloud Could that be addressed with SOP? "_Nanotrasen hires Senior staff specifically to act as mentors to new members of their departments. In addition to their regular duties, they are expected to help train anyone new to their department, upon their or others' request. Failure to do this when asked is to be considered failure to do their job, and grounds for demotion out of the department._" IE: someone takes one of these jobs, then fails to teach... and gets demoted from the job because of it. I agree that these jobs might make regular department members slightly less likely to teach, but they're unlikely anyway, and, as the saying goes, if you want something done, make sure *exactly one* person is responsible for it and accountable for ensuring it gets done.

**Comment by alexkar598** [comment]
Are sop not optional from an ooc point of view?

**Comment by Regen1** [comment]
>Am mentor and can approve. Sometimes when people ask in Mhelp about something like "How do I do Virology?" or "I'm new to Medical Doctor." It's really hard to answer those questions without showing them around as a IC character.

@ExitGame 

The issue is that being a mentor is something of your obligation so to speak, this job will have absolutely nothing entailed to it, and as fox said, will most likely end up as a special privilege job.

I'm still not sure if I like a job that just pretties out a player, if anything it should be whitelist only.

**Comment by Fox-McCloud** [comment]
That's hardly going to be an issue the vast bulk of the time; SoP is loosely enforced (or cared about), and generally only wielded against extreme "shitters" in a department. If a player does their regular responsibilities as a "senior" role, there's not going to be a ton of reason to pursue it as an SoP violation (again, especially with the low frequency of players needing a specific person in a specific department to help them out with a specific task).

And again, most shifts this job is going to be unused for its "intended" purpose; it's just going to become "job+"

**Comment by taukausanake** [comment]
I'm against karma lock for this. Better to wait for department tracking. If you do use karma a time might come when you'll have to do karma refunds. By the way, why are the karma refunds still there? No one should have those ambassador  jobs anymore

**Comment by Dinarzad** [comment]
Can someone explain to me more this "This will be job+" mindset? Because I'm having a little trouble seeing exactly how that'll come about.

To my understanding this is a'job-' It's the same job but with MORE work on top of it, as you're expected to teach people. 
If you nixed the bit about enouraging command to pick a senior role when picking an Acting Head, then all 'privelage' about it would be lost, no? You're literally just an IC MEntor at that point, no special perks just a broadcasted willingness to take extra work.

**Comment by Anticept** [comment]
@SamHPurp, I actually agree with @Fox-McCloud on this. It will be abused to get past the job locks, and I'm not so sure SOP is the answer.

I think these should be WHITELISTED roles. I also don't really agree with the "Senior" prefix. Perhaps these would be better descriptors?

- Professor of Engineering.
- Professor of Medicine.
- Academy Instructor (security).
- Research Professor

**Comment by taukausanake** [comment]
If you go with Professor then these are hired professionals not part of station crew and kind of changes things a bit. They are in a sense special NT members with less rights. Sounds like mentor rolls now

**Comment by SheepFighter** [comment]
I have mixed feelings on this. Obviously, the fewer greyshirts the better. But I have a nasty suspicion that most crewmembers won't bother taking advantage of this.

**Comment by Doukhan** [comment]
This would only encourage snowflaking and "we're just better" attitudes that we don't need anymore of.

**Comment by tylerh95** [comment]
+1 really like this idea, will benefit new players

Also, special fluff gear would be cool.

**Comment by Regen1** [comment]
Instead of making it a separate job, could you add it so there's a toggleable button next to the job?

It's be like "Be senior? Yes / No"

So those who have it unlocked can toggle if they want to spawn as a senior or not, if yes they'll be put into the senior "pick pool", if they don't get a senior role they'll just be thrown back into the normal job picking poll.

**Comment by ExitGame** [comment]
Really, there is something wrong if we cannot add features because some people could abuse with being Elitist with it. You should just hammer on them down with IC justice, there's a lot of options.

**Comment by tylerh95** [comment]
> Really, there is something wrong if we cannot add features because some people could abuse with being Elitist with it. You should just hammer on them down with IC justice, there's a lot of options.

This 100%. The mindset of "this can be abused" or "people won't use as intended" instead of simply giving it a shot is a huge limitation on this servers potential.

**Comment by Lady-Luck** [comment]
That's just being flat-out ignorant of human nature. If you design your game in a way that encourages certain behavior, you're going to see that behavior.

**Comment by tylerh95** [comment]
> That's just being flat-out ignorant of human nature. If you design your game in a way that encourages certain behavior, you're going to see that behavior.

I never said it couldn't be reverted. I'm simply saying we shouldn't be scared of trying features like this because of what "might" happen.

**Comment by Lady-Luck** [comment]
>I never said it couldn't be reverted. I'm simply saying we shouldn't be scared of trying features like this because of what "might" happen.

While I agree with this mentality for most things, but this is tying into the karma system. Doing a new round of karma refunds if this doesn't work out is not ideal.

**Comment by tylerh95** [comment]
> While I agree with this mentality for most things, but this is tying into the karma system. Doing a new round of karma refunds if this doesn't work out is not ideal.

Agreed, I'm not sure why refunds cannot be handled by a system automatically with SQL queries.

**Comment by tylerh95** [comment]
> While I agree with this mentality for most things, but this is tying into the karma system. Doing a new round of karma refunds if this doesn't work out is not ideal.

If our refund system is such a pain, then we really should be considering ways to improve that as a system. If the argument against new karma locked features is regarding the actual karma system and not the suggested feature, then we really ought to fix our karma system.

**Comment by Lady-Luck** [comment]
>If our refund system is such a pain, then we really should be considering ways to improve that as a system. If the argument against new karma locked features is regarding the actual karma system and not the suggested feature, then we really ought to fix our karma system.

But that's besides the point. This isn't people not wanting to try new things, i.e. it's not the idea of senior roles being argued against. The departmental tracking is just a better system for implementation.

**Comment by tylerh95** [comment]
> But that's besides the point. This isn't people not wanting to try new things, i.e. it's not the idea of senior roles being argued against. The departmental tracking is just a better system for it.

I have to disagree. I think that if it is simply based on time played, it's more likely to become "just another slot". If someone has to use karma they've saved for that position, I think it'd make it more likely that people who actually want to play the role will end up in the role.

**Comment by HereticChurch** [comment]
I think the best option for this idea would be to have most primary jobs have "Senior" as an alt title selectable by anyone. No karma lock or whitelist or extra job slot (Maybe a time lock? Can we do that for titles?). Just a way to say "I am more experienced and might be more willing to help" without getting any special treatment or perks for it outside of your ego being stroked ever so slightly by a fancier (meaningless) title.

**Comment by Anticept** [comment]
It would lose the meaning of the title. In a way, it should be the way mentors are treated. We are volunteers who want to make it better. Someone with a senior title needs to earn it by demonstrating the same attitude.

**Comment by Vivalas** [comment]
I for once actually agree with Fox and the mindset of designing a feature around the way it could possibly be abused. While I love the idea I feel like it will just end up messy. It's worth a shot though (and it feels like something relevant I can actually spend my karma on rather than meme jobs)

File: code/game/jobs/job/engineering.dm [hunk 1]
```diff
@@ -43,6 +43,40 @@
...
+	supervisors = "the chief engineer"
+	selection_color = "#fff5cc"
+	access = list(access_eva, access_engine, access_engine_equip, access_tech_storage, access_maint_tunnels, access_external_airlocks, access_construction, access_atmospherics, access_mineral_storeroom)
+	minimal_access = list(access_eva, access_engine, access_engine_equip, access_tech_storage, access_maint_tunnels, access_external_airlocks, access_construction, access_mineral_storeroom)
+	minimal_player_age = 30
+	exp_requirements = 3400 // This'll be changed to be karma related once I've worked that bit out.
+	exp_type = EXP_TYPE_CREW
+	outfit = /datum/outfit/job/senioreng
+
+/datum/outfit/job/senioreng // Since nothing actually changes, equipment-wise, could probably just use the regular datum
```
Comment (tigercat2000) [inline code/game/jobs/job/engineering.dm]: You definitely should infact use the regular datum if nothing's changed and you don't plant to change anything.

File: code/game/jobs/job/medical.dm [hunk 2]
```diff
@@ -39,6 +39,38 @@
...
+	supervisors = "the chief medical officer"
+	selection_color = "#ffeef0"
+	access = list(access_medical, access_morgue, access_surgery, access_chemistry, access_virology, access_genetics, access_mineral_storeroom)
+	minimal_access = list(access_medical, access_morgue, access_surgery, access_maint_tunnels)
+	minimal_player_age = 30
+	exp_requirements = 3400 // This'll be changed to be karma related once I've worked that bit out.
+	exp_type = EXP_TYPE_CREW
+	outfit = /datum/outfit/job/seniormed
+
+/datum/outfit/job/seniormed // Since nothing actually changes, equipment-wise, could probably just use the regular datum
```
Comment (tigercat2000) [inline code/game/jobs/job/medical.dm]: Same here.

File: code/game/jobs/job/security.dm [hunk 3]
```diff
@@ -89,7 +89,45 @@
...
+	supervisors = "the head of security"
+	selection_color = "#ffeeee"
+	access = list(access_security, access_sec_doors, access_brig, access_court, access_maint_tunnels, access_morgue, access_weapons)
+	minimal_access = list(access_security, access_sec_doors, access_brig, access_court, access_maint_tunnels, access_weapons)
+	minimal_player_age = 30
+	exp_requirements = 3000 // This'll be changed to be karma related once I've worked that bit out.
+	exp_type = EXP_TYPE_CREW
+	outfit = /datum/outfit/job/seniorsec
+
+/datum/outfit/job/seniorsec // Since nothing actually changes, equipment-wise, could probably just use the regular datum
```
Comment (tigercat2000) [inline code/game/jobs/job/security.dm]: You get the idea.

File: code/game/jobs/job/engineering.dm [hunk 4]
```diff
@@ -43,6 +43,40 @@
 	dufflebag = /obj/item/weapon/storage/backpack/duffel/engineering
 	box = /obj/item/weapon/storage/box/engineer
 
+/datum/job/senioreng
```
Comment (tigercat2000) [inline code/game/jobs/job/engineering.dm]: Most of these variables are the same, right? Why not just make all of these senior roles a subtype of the job itself? As long as they have unique flags, I can't think of any reason that could go wrong.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 5]
```diff
@@ -644,7 +644,7 @@
 						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-med"), ICON_OVERLAY)
 					if(4)
 						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-			if(DOCTOR)
+			if(DOCTOR || SENIORMED)
```
Comment (tigercat2000) [inline code/modules/mob/new_player/preferences_setup.dm]: Not related to the PR, but we should really port /tg/ and bay's update to get rid of this awful list.

Comment (SamHPurp) [inline code/modules/mob/new_player/preferences_setup.dm]: I'll take a peek at their implementation in the future for this.

File: code/game/jobs/job/engineering.dm (hunk 1 above)
Comment (SamHPurp) [inline code/game/jobs/job/engineering.dm]: I was thinking this as I did it. Thanks for the validation.

File: code/game/jobs/job/engineering.dm (hunk 4 above)
Comment (SamHPurp) [inline code/game/jobs/job/engineering.dm]: I'll test it out. Whats the worst that can happen. ;)

**Comment by Crazylemon64** [comment]
While the base idea sounds fine, the biggest issue is extra job slots that are locked by karma - meaning that people with high karma could get around a medbay population flood that'd filter out people who didn't shell out the karma. Modifying this to an alt-title may be a better policy here.

**Comment by SamHPurp** [comment]
@Crazylemon64 With a low barrier of five karma + playtime, I don't know if that'd really be a barrier to most people.

With there being other PRs up that make it easier to add an additional job slot to all of these departments already up, in the form of @Kyep 's - there doesn't seem to be a conflict there, if anything - it helps!

**Comment by SamHPurp** [comment]
While I still like this idea, there isn't the appetite for this to exist from the people whose opinion matters. 

💀

**Pull request closed without merging**